pkg/gardenlet/bootstrap/bootstrap_test.go

```diff
@@ -31,39 +35,129 @@ import (
+		c = mockclient.NewMockClient(ctrl)
 	})
 
-	Describe("#DeleteBootstrapAuth", func() {
-		var (
-			ctrl *gomock.Controller
-			c    *mockclient.MockClient
+	AfterEach(func() {
+		ctrl.Finish()
+	})
 
-			ctx     = context.TODO()
-			csrName = "csr-name"
-			csrKey  = kutil.Key(csrName)
+	Describe("#BootstrapKubeconfig", func() {
+		var (
+			seedName = "test"
+			seed     gardencorev1beta1.Seed
+
+			seedClient            client.Client
+			bootstrapClientConfig *rest.Config
+
+			gardenClientConnection *config.GardenClientConnection
+			seedConfig             config.SeedConfig
+
+			approvedCSR = certificatesv1beta1.CertificateSigningRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "watched-csr",
+				},
+				Status: certificatesv1beta1.CertificateSigningRequestStatus{
+					Conditions: []certificatesv1beta1.CertificateSigningRequestCondition{
+						{
+							Type: certificatesv1beta1.CertificateApproved,
+						},
+					},
+					Certificate: []byte("my-cert"),
+				},
+			}
+
+			deniedCSR = certificatesv1beta1.CertificateSigningRequest{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "watched-csr",
+				},
+				Status: certificatesv1beta1.CertificateSigningRequestStatus{
+					Conditions: []certificatesv1beta1.CertificateSigningRequestCondition{
+						{
+							Type: certificatesv1beta1.CertificateDenied,
+						},
+					},
+				},
+			}
 		)
 
 		BeforeEach(func() {
-			ctrl = gomock.NewController(GinkgoT())
-			c = mockclient.NewMockClient(ctrl)
+			secretReference := corev1.SecretReference{
+				Name:      "bootstrap-kubeconfig",
+				Namespace: "garden",
+			}
+
+			// gardenClientConnection with required bootstrap secret kubeconfig secret
+			// in a non-test environment we would use two different secrets
+			gardenClientConnection = &config.GardenClientConnection{
+				BootstrapKubeconfig: &secretReference,
+				KubeconfigSecret:    &secretReference,
+			}
+
+			// seedConfig containing the seed name
+			seed = gardencorev1beta1.Seed{
+				ObjectMeta: metav1.ObjectMeta{Name: seedName},
+			}
+			seedConfig = config.SeedConfig{Seed: seed}
+
+			// create mock seed client
+			s := runtime.NewScheme()
+			Expect(corev1.AddToScheme(s)).ToNot(HaveOccurred())
+			Expect(gardencorescheme.AddToScheme(s)).ToNot(HaveOccurred())
+			seedClient = fakeclient.NewFakeClientWithScheme(s, &seed)
+
+			// rest config for the bootstrap client
+			bootstrapClientConfig = &rest.Config{Host: "testhost", TLSClientConfig: rest.TLSClientConfig{
+				Insecure: false,
+				CAFile:   "filepath",
+			}}
+
+		})
+
+		It("should not return an error", func() {
+			bootstrapClient := fake.NewSimpleClientset(&approvedCSR).CertificatesV1beta1()
+
+			kubeconfig, csrName, seedName, err := BootstrapKubeconfig(ctx, testLogger, seedClient, bootstrapClient, bootstrapClientConfig, gardenClientConnection, &seedConfig)
+
+			Expect(kubeconfig).ToNot(BeNil())
+			Expect(csrName).ToNot(BeNil())
+			Expect(seedName).To(Equal(seedName))
+			Expect(err).To(BeNil())
```

Comment:
Please don't use `BeNil` for errors, `HaveOccurred` also checks if `actual` is an `error` to avoid programming tests checking the wrong stuff.
```suggestion
			Expect(err).NotTo(HaveOccurred())
```